WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/ProductCreationAIEligibilityChecker.swift

```diff
@@ -20,6 +20,12 @@ final class ProductCreationAIEligibilityChecker: ProductCreationAIEligibilityChe
             return false
         }
 
-        return site.isWordPressComStore || site.isAIAssistantFeatureActive
+        if ProcessConfiguration.shouldRespectScreenshotMode {
+            // Do not show AI product creation option in Screenshot Mode.
+            return false
+        }
```

Comment:
Alternately, we could update the mocks to make the store reflect one that is not eligible, using the checks already in the `ProductCreationAIEligibilityChecker` (not a WordPress.com store, AI assistant feature not active).

Reply:
Good point. I was trying not to change the mocked site fearing that it might create unintended breakage elsewhere. But in https://github.com/woocommerce/woocommerce-ios/pull/14335/commits/e80c6380074ec4d4d0602195b6b48038e54ddf7a I set `isWordPressComStore` as false and seems like it's not causing issue anywhere else.